### PR TITLE
Include a deprecation notice in .packages

### PR DESCRIPTION
**Pull request by sigurdm**

### CODE REVIEW

File: lib/src/lock_file.dart [hunk 1]
```diff
@@ -219,7 +219,14 @@ class LockFile {
...
   String packagesFile(SystemCache cache, [String entrypoint]) {
-    var header = 'Generated by pub on ${DateTime.now()}.';
+    var header = '''
+This file is deprecated. Tools should instead consume 
+`.dart_tools/package_config.json`.
+
+For more info see:
+https://github.com/dart-lang/language/blob/master/accepted/future-releases/language-versioning/package-config-file-v2.md
+
+Generated by pub on ${DateTime.now()}.''';
```
Comment (jonasfj) [inline lib/src/lock_file.dart:229]: Check the parser, I don't think older versions supports multiple lines of comments :see_no_evil: 

I might be remembering incorrectly...

We can test using a version of Dart 2.0.0 if we have to..

Comment (jonasfj) [reply lib/src/lock_file.dart:229]: NVM, I'm wrong :D

File: lib/src/lock_file.dart [hunk 2]
```diff
@@ -219,7 +219,14 @@ class LockFile {
   /// If [entrypoint] is passed, a relative entry is added for its "lib/"
   /// directory.
   String packagesFile(SystemCache cache, [String entrypoint]) {
-    var header = 'Generated by pub on ${DateTime.now()}.';
+    var header = '''
+This file is deprecated. Tools should instead consume 
+`.dart_tools/package_config.json`.
+
+For more info see:
+https://github.com/dart-lang/language/blob/master/accepted/future-releases/language-versioning/package-config-file-v2.md
```
Comment (jonasfj) [inline lib/src/lock_file.dart:227]: How about making a `dart.dev/go/...` link? I'm not sure this URL is very stable.

**Review by jonasfj — APPROVED** [review]

File: lib/src/lock_file.dart (hunk 2 above)
Comment (sigurdm) [reply lib/src/lock_file.dart]: https://github.com/dart-lang/site-www/pull/2888

**Pull request merged**